homeassistant/components/saj/config_flow.py

```diff
@@ -0,0 +1,94 @@
+"""Config flow for SAJ."""
+
+from __future__ import annotations
+
+from collections.abc import Mapping
+import logging
+from typing import Any
+
+import pysaj
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.exceptions import ConfigEntryAuthFailed, Unauthorized
+
+from .const import CONNECTION_TYPES, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SAJConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for the SAJ Solar Inverter."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def _async_validate_input(
+        self, user_input: dict[str, Any]
+    ) -> Mapping[str, Any]:
+        """Validate the user input allows us to connect."""
+
+        host = user_input["host"]
+        connection_type = user_input["type"]
+        username = user_input.get("username")
+        password = user_input.get("password")
+        try:
+            kwargs = {}
+            if connection_type == CONNECTION_TYPES[0]:  # Assuming 'ethernet'
+                saj = pysaj.SAJ(host)
+            else:  # Assuming 'wifi'
+                if username:
+                    kwargs["username"] = username
+                if password:
+                    kwargs["password"] = password
+                saj = pysaj.SAJ(host, **kwargs)
```

Comment:
How does this work? Can we detect this? As in, from what I understand we have ethernet and wifi as possible option, but what if we just asked the user for a host, and then just probed if we can connect, and if we detect that we need auth, we can ask the user for username and password. this way if someone doesn't need it, we are just fine!